### PR TITLE
Use the Sandbox API endpoints when mode is test

### PR DESCRIPTION
**Pull request by jhoule86**
- Make URLHelper check with ConfigHelper to choose between API and Sandbox
- Make URLHelperContract
- Hide URLHelper behind URLHelperContract
- Update modules to use URLHelperContract instead of URLHelper
- Convert AbstractConfigHelper to ConfigHelperContract
- Update modules to use ConfigHelperContract instead of 
- Register URLHelper as a singleton for the Plentymarkets app, implementing URLHelperContract
- Re-register ConfigHelper as a singleton for the Plentymarkets app, implementing ConfigHelperContract.

### CODE REVIEW

File: src/Services/InventoryUpdateService.php [hunk 1]
```diff
@@ -184,13 +185,34 @@ public function sync(bool $fullInventory = false): array
...
+      {
+        // truncate the stack to avoid PM saying the log message is too large
+        $stack = array_slice($stack, 0, 2);
+        $stack[] = '...';
+      }
+
+      if ($lenMsg > 64)
+      {
+        // message is over 300k here!
+        $msg = substr($msg, 0, 64);
```
Comment (dFed17) [inline src/Services/InventoryUpdateService.php:204]: not an issue: I'm curious if there'd be a way to encapsulate message length within the logger contract classes

Comment (jhoule86) [reply src/Services/InventoryUpdateService.php:204]: yeah that's some temporary stuff I need to clean out when I fix the merge conflicts.
Ideally, our wrapper for the PM logger can detect things that are too large and truncate and/or chunk the message.

Comment (jhoule86) [reply src/Services/InventoryUpdateService.php:230]: created https://projecthub.service.csnzoo.com/browse/EM-100/ to capture that effort

File: src/Core/Api/Services/AuthService.php [hunk 2]
```diff
@@ -28,125 +51,291 @@ class AuthService implements AuthenticationContract {
...
+   */
+  private function wayfairAuthenticate(string $audience)
+  {
+    // auth URL is the same for all Wayfair audiences.
+    $method = 'post';
+    $client_id = $this->configHelperContract->getWayfairClientId();
+    $client_secret = $this->configHelperContract->getWayfairClientSecret();
+
+    $headersArray = [
+      'Content-Type' => 'application/json',
```
[the remark this answers is not in the document]
Comment (jhoule86) [reply src/Core/Api/Services/AuthService.php:132]: yeah I'm used to that sort of thing as well - candidate for future work.

**Comment by jhoule86** [comment]
closing PR for having too many commits

**Pull request closed without merging**